### PR TITLE
Reportview tests - Tasks/10429

### PR DESCRIPTION
**Pull request by FuriousLlama**
# Fixes or Implements [AB#10429](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/10429)

-   [x] Enhancement
-   [ ] Bug
-   [ ] Documentation

## Description

Report view unit tests 

If you are reviewing this PR, please adhere to the guidelines as [documented](https://github.com/bcgov/healthgateway/wiki/prguidance).

## Testing

-   [x] Unit Tests Updated
-   [ ] Functional Tests Updated
-   [ ] Not Required

### CODE REVIEW

File: Apps/WebClient/src/ClientApp/test/timeline.test.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ function createWrapper(options?: GatewayStoreOptions): Wrapper<TimelineView> {
     });
 }
 
-describe("Timeline view", () => {
+describe("Report View", () => {
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/test/timeline.test.ts:42]: This seems like an unintentional change which should be reverted.

File: Apps/WebClient/src/ClientApp/test/reports.test.ts [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+import VueRouter from "vue-router";
+import Vuex from "vuex";
+
+import { SERVICE_IDENTIFIER } from "@/plugins/inversify";
+import { ILogger } from "@/services/interfaces";
+import { GatewayStoreOptions } from "@/store/types";
+import ReportsView from "@/views/reports.vue";
+
+import { StoreOptionsStub } from "./stubs/store/store";
+import container from "@/plugins/inversify.container";
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/test/reports.test.ts:13]: This file isn't linted.  The imports aren't sorted and there are uses of `var` instead of `const`/`let`.

**Review by MikeLyttle — CHANGES_REQUESTED** [review]

File: Apps/WebClient/src/ClientApp/test/reports.test.ts (hunk 2 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/test/reports.test.ts]: all clean now!

File: Apps/WebClient/src/ClientApp/test/timeline.test.ts (hunk 1 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/test/timeline.test.ts]: good catch. Its fixed now

File: Apps/WebClient/src/ClientApp/test/reports.test.ts [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+        stubs: { ...defaultStubs, ...customStubs },
+    });
+}
+
+describe("Report view", () => {
+    const logger: ILogger = container.get(SERVICE_IDENTIFIER.Logger);
+    logger.initialize("info");
+
+    const reportIdTag = "#reportType";
+    const testIdExclussion = "[data-testid=medicationExclusionFilter]";
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/test/reports.test.ts:82]: Typo: should be `testIdExclusion`.

File: Apps/WebClient/src/ClientApp/test/reports.test.ts [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+    const testIdSelectedDates = "[data-testid=selectedDatesFilter]";
+    const testIdClearFilter = "[data-testid=clearFilter]";
+
+    test("is a Vue instance", () => {
+        const wrapper = createWrapper();
+        expect(wrapper).toBeTruthy();
+    });
+
+    test("Availiable report types - single", () => {
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/test/reports.test.ts:91]: Typo: should be `Available`.

File: Apps/WebClient/src/ClientApp/test/reports.test.ts [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+        // Check values
+        expect(
+            wrapperSingleReport.find(reportIdTag).props()["options"].length
+        ).toBe(2);
+        expect(
+            wrapperSingleReport.find(reportIdTag).props()["options"][1].text
+        ).toBe("Medications");
+    });
+
+    test("Availiable report types - multiple", () => {
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/test/reports.test.ts:110]: Typo: should be `Available`.

File: Apps/WebClient/src/ClientApp/test/reports.test.ts [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+
+        // Check values
+        expect(wrapper.find(testIdExclussion).isVisible()).toBe(false);
+        const comboOptions = wrapper.find(reportIdTag).findAll("option");
+        await comboOptions.at(1).setSelected();
+
+        expect(wrapper.find(testIdExclussion).isVisible()).toBe(true);
+    });
+
+    test("Advanced canel", async () => {
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/test/reports.test.ts:154]: Typo: should be `cancel`.

**Review by MikeLyttle — COMMENTED** [review]
Found some typos but otherwise looks good to merge.

File: Apps/WebClient/src/ClientApp/test/reports.test.ts (hunk 6 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/test/reports.test.ts:154]: ups fixed them.

**Review by MikeLyttle — DISMISSED** [review]

File: Apps/WebClient/src/ClientApp/package.json [hunk 7]
```diff
@@ -4,10 +4,10 @@
     "version": "0.2.0",
     "scripts": {
         "serve": "vue-cli-service serve",
-        "build": "vue-cli-service lint --max-warnings 0 --no-fix && vue-cli-service build",
```
Comment (tiagograf-AOT) [inline Apps/WebClient/src/ClientApp/package.json:7]: not part of the scope of this PR

File: Apps/WebClient/src/ClientApp/test/components/monthYearPicker.test.ts [hunk 8]
```diff
@@ -1,22 +1,22 @@
 import "@/plugins/inversify.config";
-import { createLocalVue, shallowMount } from "@vue/test-utils";
 
+import { StoreOptionsStub } from "@test/stubs/store/store";
```
Comment (tiagograf-AOT) [inline Apps/WebClient/src/ClientApp/test/components/monthYearPicker.test.ts:3]: "@test/stubs/store/store"; looks redundant to me

File: Apps/WebClient/src/ClientApp/src/views/reports.vue [hunk 9]
```diff
@@ -312,15 +312,15 @@ export default class ReportsView extends Vue {
                             <b-row>
                                 <b-col class="col-12 col-lg-4 pt-3">
                                     <label for="start-date">From</label>
-                                    <DatePickerComponent
+                                    <hg-date-picker
```
Comment (tiagograf-AOT) [inline Apps/WebClient/src/ClientApp/src/views/reports.vue:315]: why this change?

File: Apps/WebClient/src/ClientApp/test/components/navbar.header.test.ts [hunk 10]
```diff
@@ -27,7 +28,7 @@ describe("NavBar Header Component", () => {
     options.modules.user.getters.userIsRegistered = () => true;
     options.modules.user.getters.userIsActive = () => true;
 
-    let customStore = new Vuex.Store(options);
+    const customStore = new Vuex.Store(options);
```
Comment (tiagograf-AOT) [inline Apps/WebClient/src/ClientApp/test/components/navbar.header.test.ts:31]: +1 for linting

**Review by tiagograf-AOT — CHANGES_REQUESTED** [review]
piggybacking changes

File: Apps/WebClient/src/ClientApp/package.json (hunk 7 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/package.json:7]: This is required to automate the linting of test files. Making changes like https://github.com/bcgov/healthgateway/pull/2255#discussion_r624420983 much easier

File: Apps/WebClient/src/ClientApp/test/components/monthYearPicker.test.ts (hunk 8 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/test/components/monthYearPicker.test.ts:3]: Agreedd. I 've changed it to /otions

File: Apps/WebClient/src/ClientApp/src/views/reports.vue (hunk 9 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/src/views/reports.vue:315]: The automated generated stub from shallowmount cases this as ````datepickercomponent-stub```. The change make sit easier to stub since the name is not changed. This is also in line with other html component changes like the hg-button and icon

**Review by MikeLyttle — APPROVED** [review]

**Comment by tiagograf-AOT** [comment]
EXEC : error Error: No files matching 'node_modules/microevent.ts' were found. [/opt/az/_work/3/s/Apps/WebClient/src/WebClient.csproj]
